flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/LookupJoinTestPrograms.java

```diff
@@ -0,0 +1,287 @@
+                                            "+I[4, 92.61, 2, Alice, 32, San Francisco, California, 95016]",
+                                            "+I[5, 12.78, 2, Alice, 32, San Francisco, California, 95016]")
+                                    .consumedAfterRestore(
+                                            "+I[7, 17.58, 6, Joana, 54, Atlanta, Georgia, 30033]")
+                                    .build())
+                    .runSql(
+                            "INSERT INTO sink_t SELECT "
+                                    + "O.order_id, "
+                                    + "O.total, "
+                                    + "C.id, "
+                                    + "C.name, "
+                                    + "C.age, "
+                                    + "C.city, "
+                                    + "C.state, "
+                                    + "C.zipcode "
+                                    + "FROM orders_t as O "
+                                    + "JOIN customers_t FOR SYSTEM_TIME AS OF O.proc_time AS C "
+                                    + "ON O.customer_id = C.id AND C.age > 30")
+                    .build();
+
+    static final TableTestProgram LOOKUP_JOIN_PRE_FILTER =
+            TableTestProgram.of("lookup-join-pre-filter", "validates lookup join with pre filter")
+                    .setupTableSource(CUSTOMERS)
+                    .setupTableSource(ORDERS)
+                    .setupTableSink(
+                            SinkTestStep.newBuilder("sink_t")
+                                    .addSchema(SINK_SCHEMA)
+                                    .consumedBeforeRestore(
+                                            "+I[2, 100.02, 5, Jake, 42, New York City, New York, 10001]",
+                                            "+I[4, 92.61, 2, Alice, 32, San Francisco, California, 95016]",
+                                            "+I[5, 12.78, 2, Alice, 32, San Francisco, California, 95016]")
+                                    .consumedAfterRestore(
+                                            "+I[7, 17.58, 6, Joana, 54, Atlanta, Georgia, 30033]")
+                                    .build())
+                    .runSql(
+                            "INSERT INTO sink_t SELECT "
+                                    + "O.order_id, "
+                                    + "O.total, "
+                                    + "C.id, "
+                                    + "C.name, "
+                                    + "C.age, "
+                                    + "C.city, "
+                                    + "C.state, "
+                                    + "C.zipcode "
+                                    + "FROM orders_t as O "
+                                    + "JOIN customers_t FOR SYSTEM_TIME AS OF O.proc_time AS C "
+                                    + "ON O.customer_id = C.id AND C.age > 30 AND O.customer_id <> 3")
```

Comment:
Can we use a field which won't cause the filter to be pushed down to the `Orders` source? A suggestion to make the test case more different from the `LOOKUP_JOIN_FILTER_PUSHDOWN`

Reply:
Changed as suggested.